raft/node.go

```diff
@@ -224,10 +224,15 @@ func RestartNode(c *Config) Node {
 	return &n
 }
 
+type proposingMsg struct {
+	m      pb.Message
+	result chan error
+}
+
 // node is the canonical implementation of the Node interface
 type node struct {
-	propc      chan pb.Message
-	recvc      chan pb.Message
+	propc      chan proposingMsg
+	recvc      chan proposingMsg
```

Comment:
do we really need to change the recvc chan? if the proposal is redirected from other peers, the waiter is actually on another node anyway. there is no easy to propagate back the error even with this change.

probably changing the propc is good enough for the current purpose.